packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -87,13 +88,25 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [[GIDSignIn sharedInstance] signOut];
     result(nil);
   } else if ([call.method isEqualToString:@"disconnect"]) {
-    [_accountRequests insertObject:result atIndex:0];
-    [[GIDSignIn sharedInstance] disconnect];
+    if ([self setAccountRequest:result]) {
+      [[GIDSignIn sharedInstance] disconnect];
+    }
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
+- (BOOL)setAccountRequest:(FlutterResult)request {
+  if (_accountRequest != nil) {
+    request([FlutterError errorWithCode:@"concurrent-requests"
```

Comment:
Agreed, though I think that safety-net logic should exist in the engine, not individual plugins.

It looks like we're already doing that -- https://github.com/flutter/engine/blob/3211d2fca262ba348a0bbcecefeb3e1b0b832faf/shell/platform/android/io/flutter/plugin/common/MethodChannel.java#L207

(we may want to change that to catch `Throwable`)